src/query/storages/common/table-meta/src/meta/v5/segment.rs

```diff
@@ -0,0 +1,205 @@
+        buf.extend(segments_compress);
+        buf.extend(summary_compress);
+
+        Ok(buf)
+    }
+}
+
+/// we need to reuse name 'SegmentInfo' in V5
+/// to reduce changes.
+pub type LeafSegmentInfo = v4::SegmentInfo;
+
+pub enum SegmentInfo {
+    #[allow(unused)]
+    InternalSegment(InternalSegmentInfo),
+    LeafSegment(LeafSegmentInfo),
+}
+
+impl From<v3::SegmentInfo> for SegmentInfo {
+    fn from(value: v3::SegmentInfo) -> Self {
+        SegmentInfo::LeafSegment(LeafSegmentInfo::from_v3(value))
+    }
+}
+
+impl<T> From<T> for SegmentInfo
+where T: Into<v2::SegmentInfo>
+{
+    fn from(value: T) -> Self {
+        SegmentInfo::LeafSegment(LeafSegmentInfo::from_v2(value.into()))
+    }
+}
+
+impl SegmentInfo {
+    #[allow(unused)]
+    pub fn new_leaf(blocks: Vec<Arc<BlockMeta>>, summary: Statistics) -> Self {
+        Self::LeafSegment(LeafSegmentInfo {
+            format_version: LeafSegmentInfo::VERSION,
+            blocks,
+            summary,
+        })
+    }
+
+    #[allow(unused)]
+    pub fn new_internal(segments: Vec<Location>, summary: Statistics) -> Self {
+        Self::InternalSegment(InternalSegmentInfo {
+            format_version: SegmentInfo::VERSION,
+            child_segments: segments,
+            summary,
+        })
+    }
+
+    // Total block bytes of this segment.
+    #[allow(unused)]
+    pub fn total_bytes(&self) -> u64 {
+        match self {
+            Self::LeafSegment(leaf_segment_info) => {
+                leaf_segment_info.summary.uncompressed_byte_size
+            }
+            Self::InternalSegment(internal_segment_info) => {
+                internal_segment_info.summary.uncompressed_byte_size
+            }
+        }
+    }
+
+    /// Serializes the Segment struct to a byte vector.
+    ///
+    /// The byte vector contains the format version, encoding, compression, and compressed block data and
+    /// summary data. The encoding and compression are set to default values. The block data and summary
+    /// data are encoded and compressed, respectively.
+    ///
+    /// # Returns
+    ///
+    /// A Result containing the serialized Segment data as a byte vector. If any errors occur during
+    /// encoding, compression, or writing to the byte vector, an error will be returned.
+    pub fn to_bytes(&self) -> Result<Vec<u8>> {
+        match self {
+            Self::LeafSegment(leaf_segment_info) => leaf_segment_info.to_bytes(),
+            Self::InternalSegment(internal_segment_info) => internal_segment_info.to_bytes(),
+        }
+    }
+
+    pub fn from_slice(bytes: &[u8]) -> Result<Self> {
+        let segment;
+        let mut cursor = Cursor::new(bytes);
+        let SegmentHeader {
+            version,
+            encoding,
+            compression,
+            // maybe we need to modify as blocks_or_segments_size
+            blocks_size,
+            summary_size,
+        } = decode_segment_header(&mut cursor)?;
+
+        if version == LeafSegmentInfo::VERSION {
```

Comment:
Is it reasonable to use VERSION to determine whether it is LeafSegmentInfo or InternalSegment?
Maybe we need add a flag in SegmentHeader to determine when serialize and  deserialize.

Reply:
fixed, no need to change SegmentHeader struct, we can get it outside